R/derive_date_vars.R

```diff
@@ -1207,25 +1207,41 @@ compute_dtf <- function(dtc, dt) {
 #' @export
 #'
 #' @examples
-#' compute_tmf(dtc = "2019-07-18T15:25", dtm = as.POSIXct("2019-07-18T15:25:00"))
-#' compute_tmf(dtc = "2019-07-18T15", dtm = as.POSIXct("2019-07-18T15:25:00"))
-#' compute_tmf(dtc = "2019-07-18", dtm = as.POSIXct("2019-07-18"))
+#' library(lubridate)
+#'
+#' compute_tmf(dtc = "2019-07-18T15:25", dtm = ymd_hms("2019-07-18T15:25:00"))
+#' compute_tmf(dtc = "2019-07-18T15", dtm = ymd_hms("2019-07-18T15:25:00"))
+#' compute_tmf(dtc = "2019-07-18", dtm = ymd("2019-07-18"))
+#' compute_tmf(dtc = "2022-05--T00:00", dtm = ymd_hms("2022-05-15T23:59:59"))
+#' compute_tmf(dtc = "2022-05--T23:00", dtm = ymd_hms("2022-05-15T23:59:59"))
+#' compute_tmf(dtc = "2022-05--T23:59:00", dtm = ymd_hms("2022-05-15T23:59:59"))
+#'
 compute_tmf <- function(dtc,
                         dtm,
                         ignore_seconds_flag = FALSE) {
   assert_date_vector(dtm)
   assert_character_vector(dtc)
   assert_logical_scalar(ignore_seconds_flag)
 
+  valid_dtc <- is_valid_dtc(dtc)
+  warn_if_invalid_dtc(dtc, valid_dtc)
+
   partial <- get_partialdatetime(dtc)
   highest_miss <- convert_blanks_to_na(vector("character", length(dtc)))
-  for (c in c("hour", "minute", "second")) {
+  hms <- c("hour", "minute", "second")
+  time_part <-
+    map(set_names(hms), \(y) map_dbl(dtm, \(x) exec(y, x)))
```

Comment:
I really like this! It is really clever but certainly has a different look and flavor to rest of our codebase, so I left some recommended comments for the code. In addition, support of the new anonymous function short hand was introduced in purrr 1.0.0, so can you add a change to the DESCRIPTION file to include `purrr (>= 1.0.0)` ?

Reply:
I think this is an R 4.1 addition, rather than purrr 🐱

Reply:
Looks like a modification to `Depends: R (>= 4.1)` is necessary too!

Reply:
In this case, I would just update the line to 

```r
map(set_names(hms), function(y) map_dbl(dtm, function(x) exec(y, x)))
```

so we don't make an R version dependency for something that is not needed.

Reply:
Thanks @zdz2101 @ddsjoberg for the review. I removed the anonymous function shorthand and added the suggested comments.